server/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -506,75 +505,56 @@ protected Symbol visitFunctionCall(FunctionCall node, ExpressionAnalysisContext
 
         @Override
         protected Symbol visitSimpleCaseExpression(SimpleCaseExpression node, ExpressionAnalysisContext context) {
-            /* CASE caseOperand
-             *   WHEN whenOperand1 THEN result1
-             *   WHEN whenOperand2 THEN result2
-             *   ELSE default
-             * END
-             * ---
-             * operands: [eq(caseOperand, whenOperand1), eq(caseOperand, whenOperand2)]
-             * results:  [result1, result2]
-             */
-            List<WhenClause> whenClauses = node.getWhenClauses();
-            List<Symbol> operands = new ArrayList<>(whenClauses.size());
-            List<Symbol> results = new ArrayList<>(whenClauses.size());
-            Symbol caseOperand = convert(node.getOperand(), context);
+            var whenClauses = node.getWhenClauses();
+            var arguments = new ArrayList<Symbol>(whenClauses.size() * 2 + 2);
+
+            var defaultValue = node.getDefaultValue();
+            arguments.add(convert(BooleanLiteral.TRUE_LITERAL, context));
```

Comment:
I think the comment in the case function regarding the signature with (true, default, ..) could be moved to a proper javadoc on class level, and here a `// See {@link CaseFunction} for why true` comment wouldn't hurt. This is not obvious and looks confusing on a first glance.